source/common/http/subscription_impl.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include "envoy/config/subscription.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+#include "common/common/macros.h"
+#include "common/config/utility.h"
+#include "common/http/headers.h"
+#include "common/http/rest_api_fetcher.h"
+
+#include "api/base.pb.h"
+#include "google/api/annotations.pb.h"
+#include "google/protobuf/descriptor.pb.h"
+#include "google/protobuf/util/json_util.h"
+
+namespace Envoy {
+namespace Http {
+
+template <class ResourceType>
```

Comment:
Presumably this makes a REST call and then does the conversion back and forth to proto? It's a bit hard to tell unless you know the proto magic. Can we get some class level comments and maybe some more info in the PR description?